### PR TITLE
Enhance retry logic and add internal DNS check

### PR DESCRIPTION
**Pull request by bingosummer**

### CODE REVIEW

File: CustomScript/customscript.py [hunk 1]
```diff
@@ -103,37 +103,81 @@ def enable(hutil):
...
+    But if download_files fails, we will wait and retry.
     """
     hutil.exit_if_enabled()
     prepare_download_dir(hutil.get_seq_no())
-    maxRetry = 2
-    for retry in range(0, maxRetry + 1):
+    public_settings = hutil.get_public_settings()
+    if public_settings is None:
+        raise ValueError("Public configuration couldn't be None.")
+    tries = public_settings.get('tries')
```
Comment (thomas1206) [inline CustomScript/customscript.py]: Shall we use retrycount?

File: CustomScript/customscript.py [hunk 2]
```diff
@@ -103,37 +103,81 @@ def enable(hutil):
...
+    is_idns_ready = check_idns()
+    while (not is_idns_ready):
+        if retry == tries:
+            error_msg = "Internal DNS is not ready, ignore it."
+            hutil.error(error_msg)
+            waagent.AddExtensionEvent(name=ExtensionShortName,
+                                      op="CheckIDNS",
+                                      isSuccess=False,
+                                      version=Version,
+                                      message="(01306)"+error_msg)
```
Comment (thomas1206) [inline CustomScript/customscript.py]: We need not only add telemetry data for failed case, but for all cases. The reason for this is we want see when can we safely remove the iDNS check.

File: CustomScript/customscript.py (hunk 1 above)
Comment (bingosummer) [inline CustomScript/customscript.py]: Here I followed the options of the command "wget".
```
Download:
  -t,  --tries=NUMBER            set number of retries to NUMBER (0 unlimits).
  -w,  --wait=SECONDS            wait SECONDS between retrievals.
```
If we used `retrycount`, then `retrycount` should be `tries -1`.

File: CustomScript/customscript.py [hunk 3]
```diff
@@ -103,37 +103,81 @@ def enable(hutil):
...
+        is_idns_ready = check_idns()
+        retry = retry + 1
+    else:
+        msg = "Internal DNS is ready"
+        hutil.log(msg)
+        waagent.AddExtensionEvent(name=ExtensionShortName,
+                                  op="CheckIDNS",
+                                  isSuccess=True,
+                                  version=Version,
+                                  message="(01306)"+msg)
```
Comment (bingosummer) [inline CustomScript/customscript.py]: This is the successful case for checking iDNS.
We can determine when to safely remove the iDNS check by it.

File: CustomScript/customscript.py (hunk 1 above)
Comment (thomas1206) [inline CustomScript/customscript.py]: OK. Looks good.

File: CustomScript/customscript.py (hunk 3 above)
Comment (thomas1206) [inline CustomScript/customscript.py]: Like what we synced offline, please add telemetry for both downloading and iDNS retry count.

**Comment by bingosummer** [comment]
Close it and I will create a new one.

**Pull request closed without merging**